ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -76,6 +95,68 @@ type RayDashboardClient struct {
 	dashboardURL string
 }
 
+func FetchDashboardAgentURL(ctx context.Context, log *logr.Logger, cli client.Client, rayCluster *rayv1alpha1.RayCluster) (string, error) {
+	dashboardAgentService := &corev1.Service{}
+	dashboardAgentServiceName := CheckName(GenerateDashboardServiceName(rayCluster.Name))
+	if err := cli.Get(ctx, client.ObjectKey{Name: dashboardAgentServiceName, Namespace: rayCluster.Namespace}, dashboardAgentService); err != nil {
+		return "", err
+	}
+
+	log.V(1).Info("fetchDashboardAgentURL ", "dashboard agent service found", dashboardAgentService.Name)
+	// TODO: compare diff and reconcile the object. For example. ServiceType might be changed or port might be modified
+	servicePorts := dashboardAgentService.Spec.Ports
+
+	dashboardPort := int32(-1)
+
+	for _, servicePort := range servicePorts {
+		if servicePort.Name == DefaultDashboardAgentListenPortName {
+			dashboardPort = servicePort.Port
+			break
+		}
+	}
+
+	if dashboardPort == int32(-1) {
+		return "", fmtErrors.Errorf("dashboard port not found")
+	}
+
+	dashboardAgentURL := fmt.Sprintf("%s.%s.svc.cluster.local:%v",
+		dashboardAgentService.Name,
+		dashboardAgentService.Namespace,
+		dashboardPort)
+	log.V(1).Info("fetchDashboardAgentURL ", "dashboardURL", dashboardAgentURL)
```

Comment:
log.V(1).Info("fetchDashboardAgentURL ", "dashboardAgentURL", dashboardAgentURL)

Reply:
@brucez-anyscale Seem this is the key change dashboardURL -> dashboardAgentURL? I can help update it in follow up PRs and let's merge this one. (We need this change in downstream to unblock other stories..)